server/service/osquery_utils/queries.go

```diff
@@ -1415,6 +1437,47 @@ func directIngestDiskEncryptionKeyDarwin(
 	return ds.SetOrUpdateHostDiskEncryptionKey(ctx, host.ID, rows[0]["filevault_key"])
 }
 
+func directIngestDiskEncryptionKeyFileLinesDarwin(
+	ctx context.Context,
+	logger log.Logger,
+	host *fleet.Host,
+	ds fleet.Datastore,
+	rows []map[string]string,
+) error {
+	if len(rows) == 0 {
+		// assume the extension is not there
+		level.Debug(logger).Log(
+			"component", "service",
+			"method", "directIngestDiskEncryptionKeyFileLinesDarwin",
+			"msg", "no rows or failed",
+			"host", host.Hostname,
+		)
+		return nil
+	}
+
+	var hexLines []string
+	for _, row := range rows {
+		if row["encrypted"] != "1" {
+			level.Debug(logger).Log(
+				"component", "service",
+				"method", "directIngestDiskEncryptionKeyDarwin",
+				"msg", "host does not use disk encryption",
+				"host", host.Hostname,
+			)
+			return nil
```

Comment:
This is the same approach taken by the existing file_lines implementation and does nothing if the host does not use disk encryption. But should we instead try to delete the prior key (if any) to cover the case where a user turns off disk encryption?

Reply:
IIRC we added this check to fix a different [bug](https://github.com/fleetdm/fleet/issues/10786). I think it's worth filing an issue and have someone look at this carefully to not introduce any regressions.